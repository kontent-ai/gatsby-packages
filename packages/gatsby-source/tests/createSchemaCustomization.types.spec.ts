import { kontentTypesCreateSchemaCustomization } from '../src/createSchemaCustomization.types';
import { createMock } from 'ts-auto-mock';
import { Actions, CreateSchemaCustomizationArgs } from 'gatsby';
import { mocked } from 'ts-jest/dist/util/testing';

describe('kontentTypesCreateSchemaCustomization', () => {
  it('create fixed type definition', () => {
    const api = createMock<CreateSchemaCustomizationArgs>({
      actions: createMock<Actions>({
        createTypes: jest.fn(),
      }),
    });

    kontentTypesCreateSchemaCustomization(api);
    const createTypesMock = mocked(api.actions.createTypes, true);
    expect(createTypesMock).toBeCalledTimes(1);
    // Fix schema
    expect(createTypesMock.mock.calls[0][0]).toMatchSnapshot();
  });
});
