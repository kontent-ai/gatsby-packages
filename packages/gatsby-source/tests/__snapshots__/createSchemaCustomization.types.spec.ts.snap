// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kontentTypesCreateSchemaCustomization create fixed type definition 1`] = `
"# All __UPPER_CASE__  wildcards will be replaced by the configuration

type kontent_type implements Node {
  id: ID!
  system: kontent_type_system!
  elements: [kontent_type_element]
}

type kontent_type_system {
  codename: String!
  id: String!
  last_modified: Date! @dateformat
  name: String!
}

type kontent_type_element {
  codename: String
  name: String
  type: String
  taxonomy_group: String
  options: [kontent_type_element_option]
}

type kontent_type_element_option {
  name: String
  codename: String
}
"
`;
